src/app/src/pages/eval-creator/components/RunTestSuiteButton.tsx

```diff
@@ -50,31 +50,37 @@ const RunTestSuiteButton: React.FC = () => {
       const job = await response.json();
 
       const intervalId = setInterval(async () => {
-        const progressResponse = await callApi(`/eval/job/${job.id}/`);
+        try {
+          const progressResponse = await callApi(`/eval/job/${job.id}/`);
 
-        if (!progressResponse.ok) {
-          clearInterval(intervalId);
-          throw new Error(`HTTP error! status: ${progressResponse.status}`);
-        }
+          if (!progressResponse.ok) {
+            clearInterval(intervalId);
+            throw new Error(`HTTP error! status: ${progressResponse.status}`);
+          }
 
-        const progressData = await progressResponse.json();
+          const progressData = await progressResponse.json();
 
-        if (progressData.status === 'complete') {
-          clearInterval(intervalId);
-          setIsRunning(false);
+          if (progressData.status === 'complete') {
+            clearInterval(intervalId);
+            setIsRunning(false);
 
-          // TODO(ian): This just redirects to the eval page, which shows the most recent eval.  Redirect to this specific eval to avoid race.
-          navigate('/eval');
-        } else if (progressData.status === 'failed') {
-          clearInterval(intervalId);
+            // TODO(ian): This just redirects to the eval page, which shows the most recent eval.  Redirect to this specific eval to avoid race.
+            navigate('/eval');
+          } else if (['failed', 'error'].includes(progressData.status)) {
+            clearInterval(intervalId);
+            setIsRunning(false);
+            throw new Error(progressData.logs.join('\n') || 'Job failed');
+          } else {
+            const percent =
+              progressData.total === 0
+                ? 0
+                : Math.round((progressData.progress / progressData.total) * 100);
+            setProgressPercent(percent);
+          }
+        } catch (error) {
```

Comment:
as function is run inside `setInterval`, it requires additional `try...catch` inside the function as well, otherwise it will be uncaught exception